src/main/java/org/opentripplanner/routing/algorithm/mapping/RaptorPathToItineraryMapper.java

```diff
@@ -146,14 +145,19 @@ else if (pathLeg.isTransferLeg()) {
       itinerary.setTransferPriorityCost(toOtpDomainCost(optimizedPath.transferPriorityCost()));
     }
 
+    var penaltyCost = 0;
     if (accessPathLeg.access() instanceof DefaultAccessEgress ae) {
       itinerary.setAccessPenalty(ae.penalty());
+      penaltyCost += ae.penalty().cost().toSeconds();
     }
 
     if (egressPathLeg.egress() instanceof DefaultAccessEgress ae) {
-      itinerary.setAccessPenalty(ae.penalty());
+      itinerary.setEgressPenalty(ae.penalty());
+      penaltyCost += ae.penalty().cost().toSeconds();
     }
 
+    itinerary.setGeneralizedCost(toOtpDomainCost(path.c1()) - penaltyCost);
```

Comment:
Shuldn't it be `+ penaltyCost` here?

Reply:
c1 already contains the penalty and since this method now returns the cost _without_ it it needs to be subtracted, not added.